src/test/compile-fail/issue-25386.rs

```diff
@@ -8,6 +8,10 @@
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
 
+// error-pattern: in this expansion of format_args!
+// error-pattern: in this expansion of print! (defined in <std macros>)
+// field `c_object` of struct `stuff::Item` is private
```

Comment:
there really isn't a good way, actually. In this case though I'd just remove the NOTEs. the issue in question (#25386) was an ICE, not a test of diagnostics (I'd not include NOTEs unless the purpose is to test that they are emitted, personally)